airbyte-scheduler/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -269,21 +273,27 @@ private Job getJob(DSLContext ctx, long jobId) {
 
   @Override
   public Optional<Job> getLastSyncJob(UUID connectionId) throws IOException {
-    return database.query(ctx -> getJobFromResult(ctx
+    return database.query(ctx -> ctx
         .fetch(BASE_JOB_SELECT_AND_JOIN + "WHERE scope = ? AND CAST(jobs.status AS VARCHAR) <> ? ORDER BY jobs.created_at DESC LIMIT 1",
```

Comment:
the problematic query style where it would pull only 1 attempt. now we get the job id of the attempt we care about and then use a helper to make sure we pull all attempts for that job.